sql/analyzer/aggregations.go

```diff
@@ -96,9 +98,12 @@ func replaceAggregatesWithGetFieldProjections(_ *sql.Context, scope *plan.Scope,
 			case sql.Aggregation, sql.WindowAggregation:
 				newAggregates = append(newAggregates, e)
 				aggPassthrough[e.String()] = struct{}{}
-				return expression.NewGetField(
-					scopeLen+len(newAggregates)-1, e.Type(), e.String(), e.IsNullable(),
-				), transform.NewTree, nil
+				typ := e.Type()
+				// special case for sum, which should always return a float64
+				if _, ok := e.(*aggregation.Sum); ok {
```

Comment:
i see tests for count, do we do all of the other agg and window functions correctly? avg, min, max, stdev, ...etc. Seems like you checked just want to make sure these are the only special cases

Reply:
would be nice if you added this to `buildAggregateFunc` also

Reply:
`aggregation.Count` is set to `int64` return type in the rule `replaceCountStar` (`sql/analyzer/expand_stars.go`) but I could also set it here?

`aggragation.Avg` should probably be all `float64s`, will make that change

I think all others are supposed to match the expression type that's passed in